doc/decisions/transformations.md

```diff
@@ -0,0 +1,98 @@
+# Transformations
+
+## Problem
+
+Certain plugins, e.g. `rename`, perform extensive transformations on keys and keysets within Elektra.
+Those transformations include, but are not limited to:
+
+- Changing the names of the keys inbetween `kdbGet` and `kdbSet` operations
+- Changing values back and forth for normalization, e.g. `true` -> `1`, `1` -> `true`
+- Adding and removing metadata (e.g. the `spec` plugin)
+
+While these features are useful, they do create feature-interaction problems.
+More specifically, problems have been observed in conjunction with the following (overlapping) types of plugins:
+
+- notification plugins
+- plugins that do change tracking
+
+The problem, in general, can be described as: Which version of the KDB should be used for notifications/change tracking?
+
+### Observed problems with changing key names
+
+For example, the `rename` plugin supports different transformations for `get` and `set`.
+The plugin is executed in the post-storage phase for the `get` operation and the pre-storage phase for the `set` operation.
+We can instruct it to convert the keynames to lowercase in the `get` operation, and to UPPERCASE in the `set` operation.
+This results in the keys being in UPPERCASE in the configuration files, but they are presented in lowercase to other plugins and applications using the Elektra API.
+
+For example, here is a configuration file with a hypothetical format:
+
+```
+/DISPLAY/BRIGHTNESS = 100
+/DISPLAY/COLOR = #FFFFFF
```

Comment:
> and avoid repetition of rename docu.

Honestly I think it makes the problem better understandable, if I repeat the essential stuff that leads to the described problem. 

If I just link to the documentation of the rename plugin, it is not guaranteed that it still works that way in a year or two. This would make understanding the rationale behind this decision much harder then.

Reply:
I haven't had time to properly review this, but from a cursory glance I think the issue is not just that you repeat the `rename` docs. I'm with you repeating the essentials is important, because the problem must remain understandable, even if `rename` changes and even if the problem doesn't exist anymore.

In fact, I think you could even totally ignore the fact that `rename` is an actual plugin and just say "suppose there is a plugin that changes key names". The problem would still be there and since the plugin is hypothetical, it would be obvious that is not a bug in the specific implementation.

Reply:
> If I just link to the documentation of the rename plugin, it is not guaranteed that it still works that way in a year or two. 

I agree.

> In fact, I think you could even totally ignore the fact that rename is an actual plugin and just say "suppose there is a plugin that changes key names". The problem would still be there and since the plugin is hypothetical, it would be obvious that is not a bug in the specific implementation.

Yes, good solution. :+1: